include/nlohmann/detail/output/binary_writer.hpp

```diff
@@ -1294,60 +1297,86 @@ class binary_writer
     template<typename NumberType, typename std::enable_if<
                  std::is_floating_point<NumberType>::value, int>::type = 0>
     void write_number_with_ubjson_prefix(const NumberType n,
-                                         const bool add_prefix)
+                                         const bool add_prefix,
+                                         const bool use_bjdata = false)
```

Comment:
Will there ever be a place where this is defaulted, or should it always be provided so that we can't forget it?

Reply:
because the function name still has `ubjson` in it, so I thought it is helpful to assume it is the default format, while requiring all bjdata calls via this function carries the burden of using the extra parameter.

but I confirm all current presences of this function have the extra parameter and happy to remove the default.

Reply:
If all of the current callers are providing it, then I would say no default, so that we don't accidentally add one without considering the proper value in the future.

Reply:
fixed